cypress-io/cypress
# feat: Re-introduce Run All specs for End to End under experimentalRunAllSpecs flag

- Closes https://github.com/cypress-io/cypress/issues/24168

### User facing changelog


Adds a new experimental flag, `experimentalRunAllSpecs`, for End to End Testing. Enable it in `cypress.config`:

```ts
import { defineConfig } from 'cypress'

export default defineConfig({
  e2e: {
    experimentalRunAllSpecs: true
  }
})
```

### Additional details


This PR adds back in support for running multiple tests sequentially in Open Mde for End to End testing. It can be accessed from both the Specs page and the Runner page in the Cypress App:

![image](https://user-images.githubusercontent.com/19196536/202962060-e7140100-3d62-4e01-8692-ac03d034ddc4.png)

![ss](https://user-images.githubusercontent.com/19196536/202962468-30d0faea-423d-4f03-bf2e-25f77bce5750.png)


### Steps to test


1. Either check out this branch locally, or install the pre-release binary. [Those are here](https://github.com/cypress-io/cypress/commit/30163cbfb060a21cb7ccc0e424951bcc6113396b#comments).
2. Enable the experiment:

```ts
import { defineConfig } from 'cypress'

export default defineConfig({
  e2e: {
    experimentalRunAllSpecs: true
  }
})
```

Open Cypress in Open Mode. The Specs page and Runner page should be have "Run N Specs" labels that are revealed when you hover a directory. Clicking the button will re-open the browser and execute all the specs sequentially.

You can also use the text search to only execute a sub-set of specs. Saving any of the specs will re-execute all the specs that matched your criteria (search, directory, or both).

Note: It's not fully a11y yet. You cannot do Run All Specs with a keyboard only yet. I filed an issue for a fast follow: https://github.com/cypress-io/cypress/issues/24748. If there's a trivial fix, that'd be great.

### How has the user experience changed?


See above.

### PR Tasks


- [x Has a PR for user-facing changes been opened in [`cypress-documentation`](https://github.com/cypress-io/cypress-documentation)? https://github.com/cypress-io/cypress-documentation/pull/4860
